lib/pure/asyncdispatch.nim

```diff
@@ -640,45 +635,15 @@ when defined(windows) or defined(nimdoc):
 
     let ret = WSARecv(socket.SocketHandle, addr dataBuf, 1, addr bytesReceived,
                       addr flagsio, cast[POVERLAPPED](ol), nil)
-    if ret == -1:
+    # Processing of "immediate completion" operations was removed, because under
+    # heavy load it can lead to deep recursion and stack overflow.
+    # This change also dont affect speed.
+    if ret != 0:
       let err = osLastError()
       if err.int32 != ERROR_IO_PENDING:
-        if dataBuf.buf != nil:
-          dealloc dataBuf.buf
-          dataBuf.buf = nil
+        dealloc dataBuf.buf
         GC_unref(ol)
-        if flags.isDisconnectionError(err):
```

Comment:
Made some tests with windows host and windows client (with TCPView), so its a method when RST packet is sending to remote side (host).
Results when WSARecv() is not started yet:
WSARecv() returned 0, and osLastError() == 0, callback was called with bytesCount = 0 and oserror = -1
Results when WSARecv() was started but not received any data yet:
WSARecv() returned -1 and osLastError() = 997 (ERROR_IO_PENDING), callback was called with bytesCount = 0 and oserror = -1